app/src/main/java/com/vero/tasky/agenda/di/AgendaModule.kt

```diff
@@ -69,19 +80,43 @@ object AgendaModule {
 
     @Provides
     @Singleton
-    fun provideSyncAgendaWorkManagerRunner(@ApplicationContext context: Context)
+    fun provideSyncAgendaWorkerRunner(@ApplicationContext context: Context)
         : SyncAgendaWorkerRunner {
         return SyncAgendaWorkerRunnerImpl(
             workManager = WorkManager.getInstance(context)
         )
     }
 
+    @Provides
+    @Singleton
+    fun provideCreateEventWorkerRunner(@ApplicationContext context: Context)
+            : CreateEventWorkerRunner {
+        return CreateEventWorkerRunnerImpl(
+            workManager = WorkManager.getInstance(context)
```

Comment:
You can create another Provides function, that provides the workmanager

So you can just pass the instance

Mostly because you'll create another workers, so it makes sense to have the Workmanager creation in one place